services/QuillLMS/lib/tasks/schools.rake

```diff
@@ -107,4 +107,61 @@ namespace :schools do
       end
     end
   end
+
+  # A bit more detail here: when we did our original population of school
+  # data from federal NCES data sources, NCES ID values included left-
+  # padding "0"s to make them all the same length.  When we updated our
+  # data in 2022 from the same federal sources, the NCES IDs were no longer
+  # left-padded in the data set, so we ended up creating duplicate schools.
+  # One set with leading "0"s and one without.  This task is intended to
+  # move any users from the duplicates to the original schools, and then to
+  # remove the duplicates from our database.
+  desc 'Clean up duplicate schools based on 0-left-padded NCES ID'
+  task :clean_up_duplicates => :environment do
+
+    teachers_assigned_to_duplicates = <<-SQL
+      SELECT original.id AS original_school_id,
+            duplicate.id AS duplicate_school_id,
+            schools_users.id AS schools_users_id,
+            schools_admins.id AS schools_admins_id
+          FROM schools AS original
+          JOIN schools AS duplicate
+              ON original.nces_id = CONCAT('0', duplicate.nces_id)
+                  OR original.nces_id = CONCAT('0', duplicate.nces_id)
+          LEFT OUTER JOIN schools_admins
+              ON duplicate.id = schools_admins.school_id
+          LEFT OUTER JOIN schools_users
+              ON duplicate.id = schools_users.school_id
+    SQL
+
+    users_updated = 0
+    admins_updated = 0
+    duplicate_schools_deleted = 0
+    ActiveRecord::Base.connection.execute(teachers_assigned_to_duplicates).each do |row|
```

Comment:
This is more personal preference / risk tolerance, but I would migrate all data from the duplicates to the originals in one task, then do the deletions in a separate task, allowing for inspection between the two tasks.